cmonge/models/nn.py

```diff
@@ -304,24 +305,67 @@ def create_train_state(
 class ConditionalPerturbationNetwork(ModelBase):
     dim_hidden: Sequence[int] = None
     dim_data: int = None
-    dim_cond: int = None
-    dim_cond_map: int = None
+    dim_cond: int = None  # Full dimension of all context variables concatenated
+    # Same length as context_entity_bonds if embed_cond_equal is False (if True, first item is size of deep set layer, rest is ignored)
+    dim_cond_maps: Iterable[int] = (50, 1)
     act_fn: Callable[[jnp.ndarray], jnp.ndarray] = nn.gelu
     is_potential: bool = False
     layer_norm: bool = False
+    embed_cond_equal: bool = False
+    context_entity_bonds: Iterable[Tuple[int, int]] = (
+        (0, 10),
+        (0, 11),
+    )  # Start/stop index per modality
+    dim_cond_map: int = 50  # Depreciated, for backwards compatibility
```

Comment:
I dont see why this backwards compatibility is needed, we could just adapt the loader to pass items always as lists so that we do `dim_cond_maps=[50]`

Reply:
This is always passed via the config.model. The old configs all have `dim_cond_map=50` but we can catch this differently.